### PR TITLE
🧹 use cnspec directly to scan container images

### PR DESCRIPTION
**Pull request by imilchev**
Instead of using the scan API, use cnspec directly to run a container scan.

- [x] add `containers` configuration section in `MondooAuditConfig`
- [x] use `cnspec` directly for running container scans
- [x] adjust default resource limits and requirements for scan API
- [x] adjust tests

### CODE REVIEW

**Comment by czunker** [comment]
I'm currently trying to test this locally, but get errors during Pod startup:
```
  Warning  FailedMount  3m35s                 kubelet            MountVolume.SetUp failed for volume "config" : write /var/lib/kubelet/pods/f58936d5-031e-46b1-9340-f5066b40cef5/volumes/kubernetes.io~projected/config/..2023_05_31_06_07_01.2165409312/mondoo/mondoo.yml: no space left on device
  Warning  FailedMount  3m35s                 kubelet            MountVolume.SetUp failed for volume "kube-api-access-28d68" : write /var/lib/kubelet/pods/f58936d5-031e-46b1-9340-f5066b40cef5/volumes/kubernetes.io~projected/kube-api-access-28d68/..2023_05_31_06_07_01.1634284187/namespace: no space left on device
```
This seems wired. The disk still has space free, free inodes are present, ... Have to look into it.

**Comment by czunker** [comment]
I tried it locally and it works:
```
...
→ resolved assets resolved-assets=12
...
Scanned 12 assets

Alpine Linux v3.18
    U index.docker.io/mondoo/cnspec@cf956a93eaca

Distroless
    U quay.io/jetstack/cert-manager-cainjector@7c65d8478484
    U quay.io/jetstack/cert-manager-controller@63a7aa17a1b1
    U quay.io/jetstack/cert-manager-webhook@c5644d09c6cf
    U registry.k8s.io/etcd@6f72b8515449
    U registry.k8s.io/kube-apiserver@4f261d18da8d
    U registry.k8s.io/kube-controller-manager@00a7408ea4f0
    U registry.k8s.io/kube-proxy@f6718b6cb8f8
    U registry.k8s.io/kube-scheduler@4ee9dc8056c0
    U registry.k8s.io/metrics-server/metrics-server@c60778fa1c44

scratch
    X gcr.io/k8s-minikube/storage-provisioner@18eb69d1418e
    X registry.k8s.io/coredns/coredns@8e352a029d30

Summary
=======

Score Distribution		Asset Distribution
------------------		------------------
A   0 assets      		scratch              2
B   0 assets      		Alpine Linux v3.18   1
C   0 assets      		Distroless           9
D   0 assets
F   0 assets
U  10 assets
X   2 assets
...
```

The only thing I'm not sure about is that the container tries to restart because the scan does not exit with zero.
I assume we also had this before. Should we set the restart policy to `Never`?

File: tests/integration/audit_config_base_suite.go [hunk 1]
```diff
@@ -133,29 +133,45 @@ func (s *AuditConfigBaseSuite) testMondooAuditConfigKubernetesResources(auditCon
...
 	s.True(
 		s.testCluster.K8sHelper.WaitUntilCronJobsSuccessful(utils.LabelsToLabelSelector(cronJobLabels), auditConfig.Namespace),
 		"Kubernetes container image scan CronJob did not run successfully.")
 
 	err = s.testCluster.K8sHelper.CheckForPodInStatus(&auditConfig, "client-k8s-images-scan")
 	s.NoErrorf(err, "Couldn't find container image scan pod in Podlist of the MondooAuditConfig Status")
 
-	err = s.testCluster.K8sHelper.CheckForReconciledOperatorVersion(&auditConfig, version.Version)
-	s.NoErrorf(err, "Couldn't find expected version in MondooAuditConfig.Status.ReconciledByOperatorVersion")
+	// err = s.testCluster.K8sHelper.CheckForReconciledOperatorVersion(&auditConfig, version.Version)
```
[the remark this answers is not in the document]
Comment (imilchev) [reply tests/integration/audit_config_base_suite.go:173]: commented out for the tests. It should actually be enabled. I will uncomment it

File: config/crd/bases/k8s.mondoo.com_mondooauditconfigs.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.10.0
+    controller-gen.kubebuilder.io/version: v0.9.0
```
[the remark this answers is not in the document]
Comment (imilchev) [reply config/crd/bases/k8s.mondoo.com_mondooauditconfigs.yaml:6]: not sure how much it matters but I can update my local kubebuilder version

File: controllers/nodes/deployment_handler_test.go [hunk 3]
```diff
@@ -232,7 +232,7 @@ func (s *DeploymentHandlerSuite) TestReconcile_CreateCronJobs() {
 	nodes := &corev1.NodeList{}
 	s.NoError(d.KubeClient.List(s.ctx, nodes))
 
-	image, err := s.containerImageResolver.MondooClientImage(
```
[the remark this answers is not in the document]
Comment (imilchev) [reply controllers/nodes/deployment_handler_test.go:235]: there are still a few other places where mondoo client is mentioned. I will leave those for another PR though

**Review by czunker — APPROVED** [review]
Nice! Thanks @imilchev

**Pull request merged**